src/cmap/auth/mongo_credentials.ts

```diff
@@ -137,6 +150,44 @@ export class MongoCredentials {
       throw new MongoMissingCredentialsError(`Username required for mechanism '${this.mechanism}'`);
     }
 
+    if (this.mechanism === AuthMechanism.MONGODB_OIDC) {
```

Comment:
Sorry, just thought of something, will we merge this now or hold off until the implementation is ready (I think the latter is annoying process wise). 

Should we throw an error here to prevent this from appearing to be "supported" before it actually is if we ship this logic in v5.1.0?

Reply:
I tagged the fix version in the tickets for 5.1.0 since I expected it to go in that release. But I guess it would be odd if the actual spec wasn't completed. But this will error if the user tries to use it due to the auth provider not being there yet. See: https://github.com/mongodb/node-mongodb-native/blob/main/src/cmap/connect.ts#L38 and https://github.com/mongodb/node-mongodb-native/blob/main/src/cmap/connect.ts#L111

Reply:
Interesting, a runtime error seems good to me, and we can callout in release notes if that it doesn't work yet if we reach 5.1.0 without the functionality